contracts/mediator/ForeignMediator.sol

```diff
@@ -0,0 +1,43 @@
+pragma solidity 0.4.24;
+
+import "./BasicMediator.sol";
+import "../interfaces/IHomeMediator.sol";
+
+contract ForeignMediator is BasicMediator {
+    function passMessage(address _from, uint256 _tokenId) internal {
+        bytes memory metadata = getMetadata(_tokenId);
+
+        bytes4 methodSelector = IHomeMediator(0).handleBridgedTokens.selector;
+        bytes memory data = abi.encodeWithSelector(methodSelector, _from, _tokenId, metadata);
+
+        bytes32 dataHash = keccak256(data);
```

Comment:
are we sure that data will be unique for the same token? my understanding is that `dataHash` should be unique for every relay operation.

Reply:
Thanks! I think the `tokenId` confused me in this case. If the same token is bridged a second time with the same metadata it won't be unique. I think we should add a `nonce` attribute similar to the implementation of erc677 mediators

Reply:
Please check https://github.com/poanetwork/cryptokitties-xdai-demo/pull/6/commits/0c4c03b093429743d58a3632568f0b941be5857d